tuple/include/array_of_doubles_intersection.hpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef ARRAY_OF_DOUBLES_INTERSECTION_HPP_
+#define ARRAY_OF_DOUBLES_INTERSECTION_HPP_
+
+#include <vector>
+#include <memory>
+
+#include "array_of_doubles_sketch.hpp"
+#include "tuple_intersection.hpp"
+
+namespace datasketches {
+
+template<
+  typename Policy,
```

Comment:
For consistency purposes, is it better to use Mode here, or use Policy in the java version?  The same conceptual idea has different names in both implementations.  Do we strive for this or is this a non-issue?

Reply:
We try to be consistent if it makes sense. I think in this case the implementation is substantially different. A corresponding thing in Java ArrayOfDoublesIntersection is called Combiner. And there is no way currently to override similar combining in sketch update and in union. Here in C++ I decided to call this concept a Policy, and it can be user-defined for update, union and intersection. It defaults to sum in update and union, but there is no default for the intersection since it is not quite clear what to do with values for the same keys. The naming is debatable, of course.

Reply:
I agree with your reasons for keeping it as it is, seeing as the implementations are different. The default policy that you have chosen for the different operations also sensible.